### PR TITLE
DROID-2121 Design | Relations value widget

### PR DESCRIPTION
**Pull request by konstantiniiv**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+//                onQueryChanged = { s -> }
+//            )
+            Divider(paddingEnd = 0.dp, paddingStart = 0.dp)
+            RelationsLazyList(state = state, action = action)
+        }
+    }
+}
+
+@Composable
+private fun WidgetHeader(state: RelationValueViewState, action: (RelationValueViewAction) -> Unit) {
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:73]: what is "widget" here?

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        state = lazyListState,
+        modifier = Modifier
+            .fillMaxWidth()
+            .wrapContentHeight()
+    ) {
+        itemsIndexed(
+            items = state.items,
+            itemContent = { _, item ->
+                when (item) {
+//                    is RelationValueView.Create -> ItemTagOrStatusCreate(state = item)
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:172]: will be implemented later?

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+//                    is RelationValueView.Option.Status -> StatusItem(state = item)
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:173]: will be implemented later?

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+                    is RelationValueView.Option.Tag -> TagItem(state = item, action = action)
+                    else -> TODO()
+                }
+            })
+    }
+}
+
+@Composable
+fun RelationsViewEmpty() {
```
Comment (uburoiubu) [inline core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:182]: // TODO?

**Review by uburoiubu — APPROVED** [review]

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt (hunk 3 above)
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:173]: yes

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+}
+
+@Composable
+fun RelationsViewLoading() {
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:186]: added

File: presentation/src/main/java/com/anytypeio/anytype/presentation/relations/model/RelationsListItem.kt [hunk 6]
```diff
@@ -0,0 +1,30 @@
+package com.anytypeio.anytype.presentation.relations.model
+
+import com.anytypeio.anytype.core_models.ThemeColor
+
+sealed class RelationsListItem {
+
+    abstract val text: String
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/relations/model/RelationsListItem.kt:7]: no

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt (hunk 1 above)
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/RelationsLazyList.kt:73]: renamed to Header

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/Common.kt [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import com.anytypeio.anytype.core_ui.R
+import com.anytypeio.anytype.core_ui.views.PreviewTitle2Medium
+import com.anytypeio.anytype.core_ui.views.Relations1
+import com.anytypeio.anytype.presentation.relations.model.RelationsListItem
+
+@Composable
+fun CommonContainer(modifier: Modifier = Modifier, content: @Composable BoxScope.() -> Unit) {
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/Common.kt:30]: it's container for all relation value items in lists

File: core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/TagItem.kt [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+                .size(36.dp)
+                .align(Alignment.CenterEnd)
+        )
+        Divider(modifier = Modifier.align(Alignment.BottomCenter))
+    }
+}
+
+@Composable
+fun TagItemText(state: RelationValueView.Option.Tag) {
+    val themeColor = ThemeColor.values().find { it.code == state.color } ?: ThemeColor.DEFAULT
```
[the remark this answers is not in the document]
Comment (konstantiniiv) [reply core-ui/src/main/java/com/anytypeio/anytype/core_ui/relations/TagItem.kt:55]: will add

**Pull request merged**